### PR TITLE
Add replication stream for pushers

### PR DESCRIPTION
**Pull request by NegativeMjark**
Todo:
* [x] hook up adding/removing pushers to the notifier so that it wakes up replication long-polls when pushers are added.

### CODE REVIEW

File: synapse/storage/schema/delta/30/deleted_pushers.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE TABLE IF NOT EXISTS deleted_pushers(
+    stream_id BIGINT NOT NULL,
+    app_id TEXT NOT NULL,
+    pushkey TEXT NOT NULL,
+    user_id TEXT NOT NULL,
+    UNIQUE (app_id, pushkey, user_id)
```
Comment (erikjohnston) [inline synapse/storage/schema/delta/30/deleted_pushers.sql]: Do we want unique constraints if we're using this as a stream?

Comment (erikjohnston) [inline synapse/storage/schema/delta/30/deleted_pushers.sql]: Oh, I suppose you are using upsert.

Comment (NegativeMjark) [inline synapse/storage/schema/delta/30/deleted_pushers.sql]: I've added a comment to give some clue as to why there's a unique constraint  ba660ec

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**